tests/output/test_bazefetcher.py

```diff
@@ -1,69 +1,225 @@
+    t_data_start = _get_test_date(1, 13, 30)
+    t_data_end = _get_test_date(1, 15, 30)
+
+    rng = _get_test_index(t_rng_start, t_rng_end)
+    _run_bazefetcher(tmpdir, t_data_start, t_data_end, rng = rng)
+
+    expected_times = [_get_test_date(1, 14),
+                      _get_test_date(1, 15)]
+    _assert_correct_data(tmpdir, t_rng_start, t_rng_end,
+                         expected_index=expected_times)
+
+
+def test_24_hours_midnight_file(tmpdir):
+    """
+    For 24 hours time range from midnight to midnight (right border
+    opened) only 1 file is expected
+    """
+    start_day = 6
+    end_day = 7
+    t0 = _get_test_date(start_day)
+    t1 = _get_test_date(end_day)
+    ts = _run_bazefetcher(tmpdir, t0, t1)
+
+    _assert_files_list(tmpdir, start_day, start_day)
+    _assert_correctly_loaded(ts, tmpdir, t0, t1)
+
+
+def test_no_utc(tmpdir):
+    tzinfo = timezone("Australia/Melbourne")
+    tz_day = 15
+    start_hour = 8
+    end_hour = 12
+    t0 = _get_test_date(tz_day, start_hour, tzinfo=tzinfo)
+    t1 = _get_test_date(tz_day, end_hour, tzinfo=tzinfo)
+
+    series = np.array([x for x in range(start_hour, end_hour)])
+    _run_bazefetcher(tmpdir, t0, t1, series = series)
+
+
+    _assert_files_list(tmpdir, tz_day - 1, tz_day)
+    _assert_correct_data(tmpdir, t0, t1, expected_series=series)
+
+
+def test_output_data(tmpdir):
+    """
+    Tests that expected data are written to output - index and data.
+    For reference purpose uses dates with daylight saving in range
+    """
+    tzinfo = timezone("CET")
+    tz_day = 28
+    tz_month = 10
+    start_hour = 1
+    end_hour = 4
+    t0 = _get_test_date(tz_day, start_hour, month=tz_month, tzinfo=tzinfo)
+    t1 = _get_test_date(tz_day, end_hour, month=tz_month, tzinfo=tzinfo)
+
+    rng = _get_test_index(t0, t1)
+    series = list(map(lambda d : d.isoformat(), rng.to_pydatetime()))
+
+    _run_bazefetcher(tmpdir, t0, t1, series = series, rng=rng)
+
+    _assert_files_list(tmpdir, tz_day - 1, tz_day, month=tz_month)
+    _assert_correct_data(
+        tmpdir, t0, t1, expected_index=rng, expected_series=series)
+
+
+def test_no_dates_provided(tmpdir):
+    """
+    Tests bazefetcher takes full series range if no boundaries provided
+    """
+    start_day = 22
+    end_day = start_day
+    t0 = _get_test_date(start_day, 7)
+    t1 = _get_test_date(start_day, 9)
+    rng = _get_test_index(t0, t1)
+    _run_bazefetcher(tmpdir, series= np.array([1, 2]), rng=rng)
+
+    _assert_files_list(tmpdir, start_day, end_day)
+    expected_times = [_get_test_date(start_day, 7),
+                      _get_test_date(start_day, 8)]
+    _assert_correct_data(tmpdir, t0, t1, expected_index=expected_times)
+
+
+def _get_test_date(day, hour = 0, minute = 0, second = 0, year = 2018,
+                   month = 1, tzinfo = utc):
+    naive_datetime = datetime(year, month, day, hour, minute, second)
+    return tzinfo.localize(naive_datetime)
+
+
+def _get_test_index(start, end):
```

Comment:
I think you should just call the date_range function directly rather than using this helper function. Makes for fewer symbols and a more straight forward read for someone who reads the tests for examples